### DISCUSSION TITLE
How do I split shipping on a multi order

### DISCUSSION THREAD

**Discussion by dyk9456** [swellstores/community#126] [Help] [answered]
Hi 

How can I add different shipping addresses to each item in a cart and each item has a option for local pick up, local delivery or shipping

**Comment by ericingram** [comment]
It could be done using custom fields. I would probably add a new array field called `multi_shipments` with a model like this:

```
{
  "type": "array",
  "value_type": "object",
  "fields": {
    "item_ids": {
      "type": "array",
      "value_type": "objectid"
    },
    "delivery": {
      "type": "string",
      "enum": ["local_pickup", "local_delivery", "shipment"]
    },
    "shipping": {
      "type": "object",
      "extends": "shipping"
    }
  }
}
```

Then you could set up a [real-time shipping webhook](https://developers.swell.is/guides/core-concepts/webhook-basics#custom-shipping-calculation) to return a `shipment_rating` object relative to how you want to price these different options.

Integration with the Swell dashboard is another story, but maybe this quick take will help to figure it out.

**Reply by dyk9456** [reply]
Thanks you for that answer 
Would I be able to edit the cart field with that settings in the developer > console if yes can you explain how I would do that and how would I make it accessible from the fron end api

**Reply by ericingram** [reply]
You could, but easier to do it in Developer > Models. Create a new "collection" field along those lines. Actually, there's no way to add an array of objectids there, so I guess you'd have to do that in the console.

I don't have the bandwidth to explain the whole process but maybe someone else will 🙏

**Reply by dyk9456** [reply]
hi thanks again for the above code i was able to add multi_shipments but when trying to change the permission to read and write from the fronend api it dosnt work here is the body 
{
  "permissions": {
    "carts": {
      "input": {
        "fields": [
          "items",
          "items.multi_shipments",
          "items.multi_shipments.item_ids"
        ]
      },
       "fields": [
          "items",
          "items.multi_shipments",
          "items.multi_shipments.item_ids"

        ]
    }
}
}

and this is the error i am getting 
Error: Permissions for cart must be defined in api.json

**Reply by ericingram** [reply, marked as answer]
Check out this guide to enable those fields in the frontend API: https://developers.swell.is/guides/opening-content-permissions

An alternative approach is to put these fields in the `content` object, which is public by default. Just be aware that this allows anyone to modify these values in any way they want, which could break your app.

The other way to do it which doesn't include making fields public, is to use a server-side function to manage set the values. That's safer albeit takes more work.

We'll work to improve the frontend API permissions experience in the future as well.
